app/src/main/java/org/mozilla/fenix/customtabs/ExternalAppBrowserActivity.kt

```diff
@@ -74,4 +82,12 @@ open class ExternalAppBrowserActivity : HomeActivity() {
             }
         }
     }
+
+    private fun ensureSessionIsReleased() {
+        val sessionManager = components.core.sessionManager
+        val sessionId = intent.getSessionId() ?: return
+        val session = sessionManager.findSessionById(sessionId) ?: return
+
+        sessionManager.getOrCreateEngineSession(session).releaseFromView()
```

Comment:
Ideally this can be done in `BrowserStore` instead of `SessionManager`.

Reply:
@NotWoods can you give me some pointer for what you have in mind here? I can't find much `WebApp` or `Session` related that goes through the `BrowserStore`.